src/counter.rs

```diff
@@ -0,0 +1,70 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+pub struct Counter {
+    pub name: &'static str,
+    pub counts: AtomicUsize,
+    pub nanos: AtomicUsize,
+    pub times: AtomicUsize,
+    pub lograte: usize,
+}
+
+macro_rules! counter_ctor {
+    ($name:expr, $lograte:expr) => {
+        Counter {
+            name: $name,
+            counts: AtomicUsize::new(0),
+            nanos: AtomicUsize::new(0),
+            times: AtomicUsize::new(0),
+            lograte: $lograte,
+        }
+    };
+}
+
+macro_rules! counter_inc {
+    ($name:expr, $count:expr, $start:expr) => {
+        unsafe { $name.inc($count, $start.elapsed()) };
+    };
+}
+
+impl Counter {
+    pub fn inc(&mut self, events: usize, dur: Duration) {
+        let total = dur.as_secs() * 1_000_000_000 + dur.subsec_nanos() as u64;
+        let counts = self.counts.fetch_add(events, Ordering::Relaxed);
+        let nanos = self.nanos.fetch_add(total as usize, Ordering::Relaxed);
+        let times = self.times.fetch_add(1, Ordering::Relaxed);
+        if times % self.lograte == 0 && times > 0 {
+            let now = SystemTime::now().duration_since(UNIX_EPOCH).unwrap();
+            let now_ms = now.as_secs() * 1_000 + now.subsec_nanos() as u64 / 1_000_000;
+            info!(
+                "COUNTER:{{\"name:\":\"{}\", \"counts\": {}, \"nanos\": {}, \"samples\": {} \"rate\": {}, \"now\": {}}}",
```

Comment:
@sakridge should print out to a json formated blob

Reply:
Yes, I have a change that does that.